cli/src/data.ts

```diff
@@ -36,10 +37,50 @@ async function loadLusCovidData(): Promise<Dataset<DataFormat.Raw["image"]>> {
   return positive.chain(negative);
 }
 
+async function loadTinderDogData(split: number): Promise<Dataset<DataFormat.Raw["image"]>> {
+  const folder = path.join("..", "datasets", "tinder_dog", `${split + 1}`);
+  console.log(`Reading data split ${folder}`)
+  const csvPath = path.join(folder, 'labels.csv')
+
+  const headers = ['filename', 'label'];
+  const fileContent = await fs.readFile(csvPath, { encoding: 'utf-8' });
+  const csvContent = await new Promise<{ filename: string, label: number }[]>((resolve, reject) => {
+    parse(fileContent, {
+      delimiter: ',',
+      columns: headers,
+    }, (error, result: { filename: string, label: number }[]) => {
+      if (error) {
+        console.error(error);
+        reject(error)
+      }
+      resolve(result)
+    });
+  })
+  const imgToLabel = Map(csvContent.map(entry =>
+      [entry.filename, entry.label] as const)
+  );
+  const fileExtensions = [".png", ".jpg", ".jpeg"];
+  const imagesFile = (await fs.readdir(folder)).filter(file => {
+    for (const ext of fileExtensions) if(file.endsWith(ext)) return true;
+    return false;
+  })
+  const labels = imagesFile.map(img => {
+    const label = imgToLabel.get(img.slice(0, -4)) // remove the file extension
+    if (label === undefined) throw Error(`Image ${img} not found in CSV`)
+    return label.toString()
+  })
+  const imgPaths = imagesFile.map(imgName => path.join(folder, imgName))
+  console.log(`Found ${imgPaths.length} in split ${split}`)
+  const images = await Promise.all(imgPaths.map(imgPath => loadImage(imgPath)))
+  
+  return new Dataset(images).zip(labels)
+}
```

Comment:
we can use `loadCSV` and some processing to reduce the memory footprint (no need to load images before creating the dataset) and make it somewhat more readable IMO. it also drops the `csv-parse` deps from the cli.

```suggestion
function loadTinderDogData(split: number): Dataset<DataFormat.Raw["image"]> {
  const folder = path.join("..", "datasets", "tinder_dog", `${split + 1}`);

  return loadCSV(path.join(folder, "labels.csv"))
    .map(
      (row) =>
        [
          processing.extractColumn(row, "filename"),
          processing.extractColumn(row, "label"),
        ] as const,
    )
    .map(async ([filename, label]) => {
      try {
        const image = await Promise.any(
          ["png", "jpg", "jpeg"].map((ext) =>
            loadImage(path.join(folder, `${filename}.${ext}`)),
          ),
        );
        return [image, label];
      } catch {
        throw Error(`${filename} not found in ${folder}`);
      }
    });
}
```

Reply:
Ah yes using `loadCSV` is much better